qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/MigrateProcessInstanceTest.java

```diff
@@ -0,0 +1,346 @@
+                .endEvent()
+                .done());
+
+    // when
+    final var command =
+        CLIENT_RULE
+            .getClient()
+            .newMigrateProcessInstanceCommand(processInstanceKey)
+            .migrationPlan(targetProcessDefinitionKey)
+            .addMappingInstruction("A", "B")
+            .send();
+
+    // then
+    assertThatNoException()
+        .describedAs("Expect that migration command is not rejected")
+        .isThrownBy(command::join);
+
+    final var migratedProcessInstance =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.PROCESS)
+            .getFirst()
+            .getValue();
+
+    final var migratedTaskA =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+            .withRecordKey(serviceTaskKey)
+            .getFirst()
+            .getValue();
+
+    final var migratedJobA =
+        RecordingExporter.jobRecords(JobIntent.MIGRATED)
+            .withRecordKey(jobKey)
+            .getFirst()
+            .getValue();
+
+    assertThat(migratedProcessInstance)
+        .describedAs("Expect that process definition key changed")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .describedAs("Expect that bpmn process id and element id stayed the same")
+        .hasBpmnProcessId(processId)
+        .hasElementId(processId)
+        .describedAs("Expect that version number did not change")
+        .hasVersion(2);
+
+    assertThat(migratedTaskA)
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(processId)
+        .hasVersion(2)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B");
+
+    assertThat(migratedJobA)
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(processId)
+        .hasProcessDefinitionVersion(2)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B")
+        .describedAs(
+            "Expect that the type did not change even though it's different in the target process."
+                + " Re-evaluation of the job type expression is not enabled for this migration")
+        .hasType("A");
+
+    CLIENT_RULE.getClient().newCompleteCommand(jobKey).send().join();
+
+    RecordingExporter.jobRecords(JobIntent.COMPLETED).withRecordKey(jobKey).await();
+
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withElementId("sub")
+                .findAny())
+        .describedAs("Expect that sub process is activated")
+        .isPresent();
```

Comment:
🔧 IMO, the most important part about migrating a process instance is that we can continue the instance in the new process. Let's show that we can also continue the process instance in the other test cases. To do so, you could set a specific element id on the end event in the target process and check that it was activated.

Reply:
Updated!